src/channel.ts

```diff
@@ -53,12 +53,11 @@ export const ORDER = Channel.ORDER;
 export const OPACITY = Channel.OPACITY;
 
 export const CHANNELS = [X, Y, X2, Y2, ROW, COLUMN, SIZE, SHAPE, COLOR, PATH, ORDER, OPACITY, TEXT, DETAIL, LABEL];
-
-export const UNIT_CHANNELS = without(CHANNELS, [ROW, COLUMN]);
-export const UNIT_SCALE_CHANNELS = without(UNIT_CHANNELS, [X2, Y2, PATH, ORDER, DETAIL, TEXT, LABEL, X2, Y2]);
-export const SCALE_CHANNELS = UNIT_SCALE_CHANNELS.concat([ROW, COLUMN]);
-export const NONSPATIAL_CHANNELS = without(UNIT_CHANNELS, [X, Y, X2, Y2]);
-export const NONSPATIAL_SCALE_CHANNELS = without(UNIT_SCALE_CHANNELS, [X, Y]);
+export const UNIT_CHANNELS = [X, Y, X2, Y2, SIZE, SHAPE, COLOR, PATH, ORDER, OPACITY, TEXT, DETAIL, LABEL];  // CHANNELS without COLUMN, ROW
+export const UNIT_SCALE_CHANNELS = [X, Y, SIZE, SHAPE, COLOR, OPACITY]; // UNIT_CHANNELS without X2, Y2, PATH, ORDER, DETAIL, TEXT, LABEL
+export const SCALE_CHANNELS = [X, Y, SIZE, SHAPE, COLOR, OPACITY, ROW, COLUMN];  // UNIT_SCALE_CHANNELS with ROW, COLUMN
+export const NONSPATIAL_CHANNELS = [SIZE, SHAPE, COLOR, PATH, ORDER, OPACITY, TEXT, DETAIL, LABEL]; // UNIT_CHANNELS without X, Y, X2, Y2;
+export const NONSPATIAL_SCALE_CHANNELS = [SIZE, SHAPE, COLOR, OPACITY];  // UNIT_SCALE_CHANNELS without X, Y;
```

Comment:
The advantage of explicitly listing the channels is that the types are correct and so we don't have weird errors in other places.